src/watch.ts

```diff
@@ -79,20 +116,7 @@ export class Watch {
                 // ignore parse errors
             }
         });
-        stream.on('error', error);
-        stream.on('close', done);
 
-        const req = this.requestImpl.webRequest(requestOptions, (err, response, body) => {
-            if (err) {
-                error(err);
-                done();
-            } else if (response && response.statusCode !== 200) {
```

Comment:
We need to preserve this code so that we correctly handle non-200 responses.

Reply:
yes. It is done now in 'response' event handler. There we handle non-200 status codes and inject error event to the stream if the status isn't ok. Abstracting the user (watcher) from handling different error types (error on data stream, http status error), all get handled the same way then.

Reply:
thanks, I see it now.